FABulous/FABulous_CLI/FABulous.py

```diff
@@ -0,0 +1,1100 @@
+#!/usr/bin/env python
+
+# Copyright 2021 University of Manchester
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import csv
+import os
+import pickle
+import pprint
+import subprocess as sp
+import sys
+import tkinter as tk
+from contextlib import redirect_stdout
+from pathlib import Path
+
+from cmd2 import (
+    Cmd,
+    Cmd2ArgumentParser,
+    Settable,
+    categorize,
+    with_argparser,
+    with_category,
+)
+from loguru import logger
+
+from FABulous.fabric_generator.code_generation_Verilog import VerilogWriter
+from FABulous.fabric_generator.code_generation_VHDL import VHDLWriter
+from FABulous.FABulous_API import FABulous_API
+from FABulous.FABulous_CLI.helper import (
+    check_if_application_exists,
+    copy_verilog_files,
+    create_project,
+    make_hex,
+    remove_dir,
+    setup_global_env_vars,
+    setup_logger,
+    setup_project_env_vars,
+)
+
+META_DATA_DIR = ".FABulous"
+
+CMD_SETUP = "Setup"
+CMD_FABRIC_FLOW = "Fabric Flow"
+CMD_HELPER = "Helper"
+CMD_OTHER = "Other"
+CMD_GUI = "GUI"
+CMD_SCRIPT = "Script"
+CMD_OTHER = "Other"
+
+
+INTO_STRING = rf"""   
+     ______      ____        __
+    |  ____/\   |  _ \      | |
+    | |__ /  \  | |_) |_   _| | ___  _   _ ___
+    |  __/ /\ \ |  _ <| | | | |/ _ \| | | / __|
+    | | / ____ \| |_) | |_| | | (_) | |_| \__ \
+    |_|/_/    \_\____/ \__,_|_|\___/ \__,_|___/
+
+
+Welcome to FABulous shell
+You have started the FABulous shell with following options:
+{' '.join(sys.argv[1:])}
+
+Type help or ? to list commands
+To see documentation for a command type:
+    help <command>
+or
+    ?<command>
+
+To execute a shell command type:
+    shell <command>
+or
+    !<command>
+
+The shell support tab completion for commands and files
+
+To run the complete FABulous flow with the default project, run the following command:
+    load_fabric
+    run_FABulous_fabric
+    run_FABulous_bitstream ./user_design/sequential_16bit_en.v
+    run_simulation fst ./user_design/sequential_16bit_en.bin
+"""
+
+
+class FABulousShell(Cmd):
```

Comment:
To keep the structure, we might want to name the file after the class it contains.

For encapsulation, we could also rename this class/file to FABulous_CLI and add a FABulous.py in the FABulous folder that sets main().